canopeum_backend/canopeum_backend/models.py

```diff
@@ -130,6 +136,39 @@ class Coordinate(models.Model):
     dd_longitude = models.DecimalField(max_digits=9, decimal_places=6, blank=True, null=True)
     address = models.TextField(blank=True, null=True)
 
+    @staticmethod
+    def from_dms_lat_long(dms_latitude: str, dms_longitude: str):
+        dms_latitude_split = re.split(r"°|\'|\.|\"", dms_latitude)
+        dd_latitude = (
+            float(dms_latitude_split[0])
+            + (float(dms_latitude_split[1]) / 60)
+            + (float(dms_latitude_split[2] + "." + dms_latitude_split[3]) / 3600)
+        )
+        if dms_latitude_split[4] == "S":
+            dd_latitude *= -1
+
+        dms_longitude_split = re.split(r"°|\'|\.|\"", dms_longitude)
+        dd_longitude = (
+            float(dms_longitude_split[0])
+            + (float(dms_longitude_split[1]) / 60)
+            + (float(dms_longitude_split[2] + "." + dms_longitude_split[3]) / 3600)
+        )
+        if dms_longitude_split[4] == "W":
+            dd_longitude *= -1
+
+        reverse_geocode_result = gmaps.reverse_geocode(
+            (dd_latitude, dd_longitude), result_type="street_address"
+        )[0]
+        formatted_address = reverse_geocode_result["formatted_address"] if gmaps != None else ""
+
+        return Coordinate.objects.create(
```

Comment:
Goes with the classmethod comment above

```suggestion
        return cls.objects.create(
```